src/repr/src/gen/mod.rs

```diff
@@ -10,3 +10,47 @@
 //! Generated protobuf code and companion impls.
 
 pub mod row;
+
+use std::num::TryFromIntError;
+
+/// An error thrown when trying to convert from a `*.proto`-generated type
+/// `Proto$T` to `$T`.
+#[derive(Debug)]
+pub enum TryFromProtoError {
+    TryFromIntError(TryFromIntError),
+    MissingField(String),
```

Comment:
Update: saw where this got used in the next commit. I see where you're coming from here but I wonder if there's a better way to structure this. Having this variant feels like a footgun to me given the concern above.

Reply:
In practice this will never happen, but based on the structure of the generated `Proto$T` types I will always have to check that a `Proto$T` for a `$T` that is an `enum` has it's internal `oneof value` field set (I cannot generate only the inner `oneof value`). As an illustration, here is the generated structure for this PR:

```proto
syntax = "proto3";

package adt.array;

message ProtoInvalidArrayError {
    message ProtoWrongCardinality {
        uint64 actual = 1;
        uint64 expected = 2;
    }
    oneof value {
        uint64 too_many_dimensions = 1;
        ProtoWrongCardinality wrong_cardinality = 2;
    }
}
```
results in the following Rust code:

```rust
#[derive(Clone, PartialEq, ::prost::Message)]
pub struct ProtoInvalidArrayError {
    #[prost(oneof="proto_invalid_array_error::Value", tags="1, 2")]
    pub value: ::core::option::Option<proto_invalid_array_error::Value>,
}
/// Nested message and enum types in `ProtoInvalidArrayError`.
pub mod proto_invalid_array_error {
    #[derive(Clone, PartialEq, ::prost::Message)]
    pub struct ProtoWrongCardinality {
        #[prost(uint64, tag="1")]
        pub actual: u64,
        #[prost(uint64, tag="2")]
        pub expected: u64,
    }
    #[derive(Clone, PartialEq, ::prost::Oneof)]
    pub enum Value {
        #[prost(uint64, tag="1")]
        TooManyDimensions(u64),
        #[prost(message, tag="2")]
        WrongCardinality(ProtoWrongCardinality),
    }
}
```

I didn't find a way to define the `*.proto` in a way that generates the contents of `Value` in place of  `ProtoInvalidArrayError`, or to enforce `ProtoInvalidArrayError` to have a `value` field of type `Value` instead of `Option<Value>`.

Consequently, I have to handle somewhat the case that the `Option<Value>` is unset when moving from `Proto$T` to `T`. I prefer to return an error (that in principle should never occur) rather than panic.